tickets/src/main/java/no/unit/nva/publication/ticket/update/updateTicketAssigneeHandler.java

```diff
@@ -0,0 +1,139 @@
+
+public class updateTicketAssigneeHandler extends TicketHandler<TicketDto, Void> {
+
+    private final TicketService ticketService;
+    private final ResourceService resourceService;
+    private final DoiClient doiClient;
+    private static final Logger logger = LoggerFactory.getLogger(UpdateTicketStatusHandler.class);
+    public static final String COULD_NOT_CREATE_FINDABLE_DOI = "Could not create findable doi";
+    public static final String PUBLICATION_WITH_IDENTIFIER_S_DOES_NOT_SATISFY_DOI_REQUIREMENTS
+        = "Publication with identifier  %s, does not satisfy DOI requirements";
+
+    @JacocoGenerated
+    public updateTicketAssigneeHandler() {
+        this(TicketService.defaultService(),
+             ResourceService.defaultService(), new DataCiteDoiClient(HttpClient.newHttpClient(),
+                                                                     SecretsReader.defaultSecretsManagerClient(),
+                                                                     new Environment().readEnv("API_HOST")));
+    }
+
+    public updateTicketAssigneeHandler(TicketService ticketService, ResourceService resourceService,
+                                       DoiClient doiClient) {
+        super(TicketDto.class);
+        this.ticketService = ticketService;
+        this.resourceService = resourceService;
+        this.doiClient = doiClient;
+    }
+
+    @Override
+    protected Void processInput(TicketDto input, RequestInfo requestInfo, Context context) throws ApiGatewayException {
+        var ticketIdentifier = extractTicketIdentifierFromPath(requestInfo);
+        var ticket = ticketService.fetchTicketByIdentifier(ticketIdentifier);
+
+        if (userIsNotAuthorized(requestInfo, ticket)) {
+            throw new ForbiddenException();
+        }
+        if (ticket instanceof DoiRequest) {
+            doiTicketSideEffects(input, requestInfo);
+        }
+        ticketService.updateTicketAssignee(ticket, input.getAssignee());
+        return null;
+    }
+
+    @Override
+    protected Integer getSuccessStatusCode(TicketDto input, Void output) {
+        return HttpURLConnection.HTTP_ACCEPTED;
+    }
+
+    private static boolean userIsNotAuthorized(RequestInfo requestInfo, TicketEntry ticket)
+        throws UnauthorizedException {
+        return !(isAuthorizedToClaimTickets(requestInfo) && isUserFromSameCustomerAsTicket(requestInfo, ticket));
+    }
+
+    private static boolean isUserFromSameCustomerAsTicket(RequestInfo requestInfo, TicketEntry ticket)
+        throws UnauthorizedException {
+        return requestInfo.getCurrentCustomer().equals(ticket.getCustomerId());
+    }
+
+    private static boolean isAuthorizedToClaimTickets(RequestInfo requestInfo) {
+        return requestInfo.userIsAuthorized(AccessRight.APPROVE_DOI_REQUEST.toString());
+    }
+
+    private void doiTicketSideEffects(TicketDto input, final RequestInfo requestInfo)
+        throws NotFoundException, BadMethodException, BadGatewayException {
```

Comment:
From the open-api spec it seems to me that you submit a ticket with an assigne to update the database with. But from the code the assigne is fetched from the request info?

If the correct version is from request info, the open-api spec should not receive a body.

And if the body should be used so you can assign whoever you want, the findableDoiTicketSideEffects is a bit sketchy.